blue-xdma/src/include/XDMADescriptorGenerator.bsv

```diff
@@ -0,0 +1,159 @@
+import FIFOF::*;
+import Clocks :: *;
+import GetPut :: *;
+import BlueAXI :: *;
+import BlueLib :: *;
+import AXI4LiteMaster :: *;
+import StmtFSM :: *;
+`include "Config.defines"
+
+typedef struct {
+    Bit#(28) length;
+    Bit#(64) srcAddr;
+    Bit#(64) dstAddr;
+} XDMADescriptor deriving (Bits, Eq, FShow);
+
+typedef struct {
+    Bool isC2H;
+    Bool isEnable;
+} ControlDMAIfc deriving (Bits, Eq, FShow);
+
+(* always_ready, always_enabled *)
+interface IfcXDMADescriptorGeneratorFab;
+    method Bool load;
+    method Bit#(64) src_addr;
+    method Bit#(64) dst_addr;
+    method Bit#(28) len;
+    method Bit#(16) ctl;
+    (* prefix = "" *) method Action ready((* port = "ready" *) Bool r);
+endinterface
+
+interface IfcXDMADescriptorGenerator;
+    interface IfcAxi4LiteMasterFab liteFab;
+    (* prefix = "c2h_dsc_byp" *) interface IfcXDMADescriptorGeneratorFab c2hFab;
+    (* prefix = "h2c_dsc_byp" *) interface IfcXDMADescriptorGeneratorFab h2cFab;
+    interface Put#(XDMADescriptor) c2h;
+    interface Put#(XDMADescriptor) h2c;
+    method Action startC2HTransfer;
+    method Action stopC2HTransfer;
+    method Action startH2CTransfer;
+    method Action stopH2CTransfer;
+endinterface
+
+module mkXDMADescriptorGenerator(IfcXDMADescriptorGenerator);
+
+    let axi4LiteMaster <- mkAXI4LiteMaster();
+
+    FIFOF#(XDMADescriptor) c2hDescriptorFifo <- mkFIFOF;
+    FIFOF#(XDMADescriptor) h2cDescriptorFifo <- mkFIFOF;
+
+    ////////////////////////////////////////////////////////////////////////////
+    ///////////////////////   Controller   /////////////////////////////////////
+    ////////////////////////////////////////////////////////////////////////////
+
+    Reg#(AXI4_Lite_Write_Rs_Pkg) writeResponse <- mkReg(unpack(0));
+
+    Wire#(Bit#(64)) c2h_dsc_byp_src_addr <- mkDWire(0);
```

Comment:
no hard-coded literal value, use typedef for `Bit#(40)` and give it a meaningful name

Reply:
Since this is an interface exposed by XDMA IP core, the specification is fixed and does not need to be changed into a variable form.